src/main/java/frc/robot/systems/DriveFSMSystem.java

```diff
@@ -375,46 +432,109 @@ private void handleTeleOpState(TeleopInput input) {
+		// double currentEncoderPos = ((-frontLeftMotor.getEncoder().getPosition()
+		//     + frontRightMotor.getEncoder().getPosition()) / 2.0);
+		// double arcLength = (currentEncoderPos - prevEncoderPosArc)
+			// / Constants.REVOLUTIONS_PER_INCH;
+		// if (Math.abs(theta) < Constants.ODOMETRY_MIN_THETA) {
+		//     theta = Constants.ODOMETRY_MIN_THETA;
+		// }
+		// double radius = 180 * arcLength / (Math.PI * theta);
+		// double alpha = prevGyroAngle - 90;
+		// double circleX = robotXPosArc + radius * Math.cos(Math.toRadians(alpha));
+		// double circleY = robotYPosArc + radius * Math.sin(Math.toRadians(alpha));
+		// double beta = alpha + 180 - theta;
+		// robotXPosArc = circleX + radius * Math.cos(Math.toRadians(beta));
+		// robotYPosArc = circleY + radius * Math.sin(Math.toRadians(beta));
+
+		// prevGyroAngle = adjustedAngle;
+		// prevEncoderPosArc = currentEncoderPos;
+		//System.out.println("Arc: (" + robotXPosArc + ", " + robotYPosArc + ")");
+
+	}
+
+	public Point getRobotPosArc() {
+		return robotPosArc;
+	}
+
+	public Point getRobotPosLine() {
+		return robotPosLine;
+	}
+
+	private void handlePurePursuit() {
+		Point target = ppController.findLookahead();
+		if (target == null) {
+			finishedPurePursuitPath = true;
+			frontLeftMotor.set(0);
+			frontRightMotor.set(0);
+			backLeftMotor.set(0);
+			backRightMotor.set(0);
+		}
+		System.out.println("Target point: " + target.getX() + " " + target.getY());
+		Point motorSpeeds = Kinematics.inversekinematics(gyroAngle, robotPosArc, target, true);
+		frontLeftMotor.set(-motorSpeeds.getX() / 5);
+		frontRightMotor.set(motorSpeeds.getY() / 5);
+		backLeftMotor.set(-motorSpeeds.getX() / 5);
+		backRightMotor.set(motorSpeeds.getY() / 5);
+	}
+
+	private void handlePurePursuitBackward() {
+		Point target = ppController.findLookahead();
+		if (target == null) {
+			finishedPurePursuitPath = true;
+			frontLeftMotor.set(0);
+			frontRightMotor.set(0);
+			backLeftMotor.set(0);
+			backRightMotor.set(0);
 		}
-		double radius = 180 * arcLength / (Math.PI * theta);
-		double alpha = prevGyroAngle - 90;
-		double circleX = robotXPosArc + radius * Math.cos(Math.toRadians(alpha));
-		double circleY = robotYPosArc + radius * Math.sin(Math.toRadians(alpha));
-		double beta = alpha + 180 - theta;
-		robotXPosArc = circleX + radius * Math.cos(Math.toRadians(beta));
-		robotYPosArc = circleY + radius * Math.sin(Math.toRadians(beta));
-
-		prevGyroAngle = adjustedAngle;
-		prevEncoderPosArc = currentEncoderPos;
-		System.out.println("arc odo: (" + robotXPosArc + ", " + robotYPosArc + ")");
+		System.out.println("Target point: " + target.getX() + " " + target.getY());
+		Point motorSpeeds = Kinematics.inversekinematics(gyroAngle, robotPosArc, target, false);
+		frontLeftMotor.set(-motorSpeeds.getX() / 5);
+		frontRightMotor.set(motorSpeeds.getY() / 5);
+		backLeftMotor.set(-motorSpeeds.getX() / 5);
+		backRightMotor.set(motorSpeeds.getY() / 5);
```

Comment:
magic num `5`?

Reply:
We will be cleaning up the code soon (we've been procrastinating on fixing the current checkstyle errors).

Reply:
as a mentor I'm obligated to say that you should follow these best practices as you're coding, not after the fact :P please try to clean the code up before putting up a PR next time, it'll help reviewers focus on the actual code and not be confused/distracted by misc stuff :)